### PR TITLE
Port redmine issues 1179-1305

### PR DESCRIPTION
**Pull request by piponazo**
- issue 1242 commented until we solve the issue with UTF chars

### CODE REVIEW

**Comment by D4N** [comment]
I played a bit with issue 1242 and found a solution for the problem: checking the output line by line works, as the output contains at some points linefeeds instead of newlines which makes comparing it with `assertMultiLineEqual` difficult.

Also, some of the tests were making copies of the input files, but the input files were only queried and not edited, so a copy is not needed.

Please feel free to squash my commits into yours.

(Sidenote: there is also `unittest.skip()` in case you don't want to run a test, but want it in the repo)

**Comment by piponazo** [comment]
Thanks @D4N ! I think there is no need to squash your commits, in that way we keep the authorship of your changes ;).

I am just waiting for you approval to merge this.

**Review by D4N — APPROVED** [review]
I checked all the test invocations and they look the same as in bugfixes-test.sh (except for 1229, but the change is a good compromise as we don't have xmllint on windows)

**Pull request merged**
